addons/mail/static/tests/discuss/call/call_settings_menu.test.js

```diff
@@ -88,3 +91,47 @@ test("Call settings menu should not be visible on selecting a mailbox (from bein
     await contains("button[title='Hide Call Settings']", { count: 0 });
     await contains(".o-discuss-CallSettings", { count: 0 });
 });
+
+test("mobile: no key setting for push to talk", async () => {
+    mockGetMedia();
+    const pyEnv = await startServer();
+    pyEnv["discuss.channel"].create({ name: "General" });
+    mockService("discuss.ptt_extension", {
+        get isEnabled() {
+            return false;
+        },
+    });
+    patchUiSize({ size: SIZES.SM });
+    await start({ device: "mobile" });
+    await click(".o_menu_systray i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem", { text: "General" });
+    await click("[title='Start a Call']");
+    await click("[title='Open Actions Menu']");
+    await click("[title='Show Call Settings']");
+    await click("[title='toggle push-to-talk']");
+    // make sure the rendering is finished after choosing push-to-talk
+    await contains("[title='Delay after releasing push-to-talk']");
+    await contains("label[aria-label='Push-to-talk key']", { count: 0 });
+});
+
+test("mobile: do not display ptt text in settings", async () => {
+    mockGetMedia();
+    const pyEnv = await startServer();
+    pyEnv["discuss.channel"].create({ name: "General" });
+    mockService("discuss.ptt_extension", {
+        get isEnabled() {
+            return false;
+        },
+    });
+    patchUiSize({ size: SIZES.SM });
+    await start({ device: "mobile" });
+    await click(".o_menu_systray i[aria-label='Messages']");
+    await click(".o-mail-NotificationItem", { text: "General" });
+    await click("[title='Start a Call']");
+    await click("[title='Open Actions Menu']");
+    await click("[title='Show Call Settings']");
+    await click("[title='toggle push-to-talk']");
+    // make sure the rendering is finished after choosing push-to-talk
+    await contains("[title='Delay after releasing push-to-talk']");
+    await contains(".o-discuss-pttDownloadText", { count: 0 });
+});
```

Comment:
I'd drop these tests, as their value is very low: that's almost 50 additional LOCS to maintain and assertion negatives is poor.

Reply:
I think we can have these tests since they are functional important and can broke easily (a lot of condition checks inside). So that's why I added those even if we have to add something like
```
// make sure the rendering is finished after choosing push-to-talk
    await contains("[title='Delay after releasing push-to-talk']");
    ```
And the assertion negatives here is guaranteed working well.

Reply:
Slightly out-of-scope of PR, but the conditionals in template are hard not by concept but by how code was iterated in adding and adapting stuffs without some code cleaning phase.
In particular, `isMobileOS` looks too deep in the templating, it could likely be moved higher so that it's easier to see how this affects the presence of PTT extension / banner.